x-pack/legacy/plugins/apm/public/components/shared/MetadataTable/helper.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, pick, isEmpty } from 'lodash';
+import { Section as SectionType } from './sections';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { APMError } from '../../../../typings/es_schemas/ui/APMError';
+import { Span } from '../../../../typings/es_schemas/ui/Span';
+import { flattenObject, FlattenItems } from '../../../utils/flattenObject';
+
+type Item = Transaction | APMError | Span;
+
+export type MetadataItems = Array<SectionType & { data?: FlattenItems }>;
+
+export const getMetadataItems = (
+  sections: SectionType[],
+  item: Item
```

Comment:
To distinguish from a metadata item, this should be called something else. I suggest `apmDoc`. In this case I don't think you need to declare the type separately if it's not re-used.

```suggestion
  apmDoc: Transaction | APMError | Span;
```

Reply:
Thanks, I wasn't sure what would be a good name.